app/services/district_report_service.rb

```diff
@@ -69,41 +98,40 @@ def registration_counts
     SQL
   end
 
-  # We want to return cohort data for the current quarter for the selected date, and then
-  # the previous three quarters. Each quarter cohort is made up of patients registered
-  # in the previous quarter who has had a follow up visit in the current quarter.
-  def compile_cohort_trend_data
-    Quarter.new(date: selected_date).downto(3).each do |results_quarter|
-      cohort_quarter = results_quarter.previous_quarter
+  def controlled_patients_count(time)
+    ControlledPatientsQuery.call(facilities: facilities, time: time).count
+  end
 
-      period = {cohort_period: :quarter,
-                registration_quarter: cohort_quarter.number,
-                registration_year: cohort_quarter.year}
-      query = MyFacilities::BloodPressureControlQuery.new(facilities: @facilities, cohort_period: period)
-      @data[:quarterly_registrations] << {
-        results_in: format_quarter(results_quarter),
-        patients_registered: format_quarter(cohort_quarter),
-        registered: query.cohort_registrations.count,
-        controlled: query.cohort_controlled_bps.count,
-        no_bp: query.cohort_missed_visits_count,
-        uncontrolled: query.cohort_uncontrolled_bps.count
-      }.with_indifferent_access
-    end
+  def percentage(numerator, denominator)
+    return 0 if denominator == 0
+    (numerator.to_f / denominator) * 100
+  end
+
+  def top_district_quarter_stats(district)
```

Comment:
This code is not pretty, but I want to get this first version live and work on the next levels of reports before really refactoring. This will need to handle facility level as well as possibly a future state level, and that will change how to best organize this code.